src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php

```diff
@@ -109,6 +109,10 @@ public function getProperties($class, array $context = [])
      */
     public function getTypes($class, $property, array $context = [])
     {
+        if ($fromDeclaredType = $this->extractFromDeclaredType($class, $property)) {
+            return $fromDeclaredType;
+        }
+
```

Comment:
This change causing test fails:
```bash
PHPUnit 7.4.5 by Sebastian Bergmann and contributors.

Testing Symfony Property Info Component Test Suite
...............................................................  63 / 188 ( 33%)
.....SSS.................................EEEEEEEEEEEEEEEEEEEEEE 126 / 188 ( 67%)
..............................................................  188 / 188 (100%)

Time: 7.72 seconds, Memory: 8.00 MB

There were 22 errors:

1) Symfony\Component\PropertyInfo\Tests\Extractor\ReflectionExtractorTest::testExtractors with data set #0 ('a', null)
ReflectionException: Property a does not exist

/usr/src/app/src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php:205
/usr/src/app/src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php:112
/usr/src/app/src/Symfony/Component/PropertyInfo/Tests/Extractor/ReflectionExtractorTest.php:155

2) Symfony\Component\PropertyInfo\Tests\Extractor\ReflectionExtractorTest::testExtractors with data set #1 ('b', array(Symfony\Component\PropertyInfo\Type Object (...)))
ReflectionException: Property b does not exist

/usr/src/app/src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php:205
/usr/src/app/src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php:112
/usr/src/app/src/Symfony/Component/PropertyInfo/Tests/Extractor/ReflectionExtractorTest.php:155

3) Symfony\Component\PropertyInfo\Tests\Extractor\ReflectionExtractorTest::testExtractors with data set #2 ('c', array(Symfony\Component\PropertyInfo\Type Object (...)))
ReflectionException: Property c does not exist

/usr/src/app/src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php:205
/usr/src/app/src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php:112
/usr/src/app/src/Symfony/Component/PropertyInfo/Tests/Extractor/ReflectionExtractorTest.php:155

4) Symfony\Component\PropertyInfo\Tests\Extractor\ReflectionExtractorTest::testExtractors with data set #3 ('d', array(Symfony\Component\PropertyInfo\Type Object (...)))
ReflectionException: Property d does not exist

/usr/src/app/src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php:205
/usr/src/app/src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php:112
/usr/src/app/src/Symfony/Component/PropertyInfo/Tests/Extractor/ReflectionExtractorTest.php:155

5) Symfony\Component\PropertyInfo\Tests\Extractor\ReflectionExtractorTest::testExtractors with data set #4 ('e', null)
ReflectionException: Property e does not exist

/usr/src/app/src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php:205
/usr/src/app/src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php:112
/usr/src/app/src/Symfony/Component/PropertyInfo/Tests/Extractor/ReflectionExtractorTest.php:155

6) Symfony\Component\PropertyInfo\Tests\Extractor\ReflectionExtractorTest::testExtractors with data set #5 ('f', array(Symfony\Component\PropertyInfo\Type Object (...)))
ReflectionException: Property f does not exist

/usr/src/app/src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php:205
/usr/src/app/src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php:112
/usr/src/app/src/Symfony/Component/PropertyInfo/Tests/Extractor/ReflectionExtractorTest.php:155

7) Symfony\Component\PropertyInfo\Tests\Extractor\ReflectionExtractorTest::testExtractors with data set #6 ('donotexist', null)
ReflectionException: Property donotexist does not exist

/usr/src/app/src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php:205
/usr/src/app/src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php:112
/usr/src/app/src/Symfony/Component/PropertyInfo/Tests/Extractor/ReflectionExtractorTest.php:155

8) Symfony\Component\PropertyInfo\Tests\Extractor\ReflectionExtractorTest::testExtractors with data set #7 ('staticGetter', null)
ReflectionException: Property staticGetter does not exist

/usr/src/app/src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php:205
/usr/src/app/src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php:112
/usr/src/app/src/Symfony/Component/PropertyInfo/Tests/Extractor/ReflectionExtractorTest.php:155

9) Symfony\Component\PropertyInfo\Tests\Extractor\ReflectionExtractorTest::testExtractors with data set #8 ('staticSetter', null)
ReflectionException: Property staticSetter does not exist

/usr/src/app/src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php:205
/usr/src/app/src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php:112
/usr/src/app/src/Symfony/Component/PropertyInfo/Tests/Extractor/ReflectionExtractorTest.php:155

10) Symfony\Component\PropertyInfo\Tests\Extractor\ReflectionExtractorTest::testExtractors with data set #9 ('self', array(Symfony\Component\PropertyInfo\Type Object (...)))
ReflectionException: Property self does not exist

/usr/src/app/src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php:205
/usr/src/app/src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php:112
/usr/src/app/src/Symfony/Component/PropertyInfo/Tests/Extractor/ReflectionExtractorTest.php:155

11) Symfony\Component\PropertyInfo\Tests\Extractor\ReflectionExtractorTest::testExtractors with data set #10 ('realParent', array(Symfony\Component\PropertyInfo\Type Object (...)))
ReflectionException: Property realParent does not exist

/usr/src/app/src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php:205
/usr/src/app/src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php:112
/usr/src/app/src/Symfony/Component/PropertyInfo/Tests/Extractor/ReflectionExtractorTest.php:155

12) Symfony\Component\PropertyInfo\Tests\Extractor\ReflectionExtractorTest::testExtractPhp7Type with data set #0 ('foo', array(Symfony\Component\PropertyInfo\Type Object (...)))
ReflectionException: Property foo does not exist

/usr/src/app/src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php:205
/usr/src/app/src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php:112
/usr/src/app/src/Symfony/Component/PropertyInfo/Tests/Extractor/ReflectionExtractorTest.php:180

13) Symfony\Component\PropertyInfo\Tests\Extractor\ReflectionExtractorTest::testExtractPhp7Type with data set #1 ('bar', array(Symfony\Component\PropertyInfo\Type Object (...)))
ReflectionException: Property bar does not exist

/usr/src/app/src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php:205
/usr/src/app/src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php:112
/usr/src/app/src/Symfony/Component/PropertyInfo/Tests/Extractor/ReflectionExtractorTest.php:180

14) Symfony\Component\PropertyInfo\Tests\Extractor\ReflectionExtractorTest::testExtractPhp7Type with data set #2 ('baz', array(Symfony\Component\PropertyInfo\Type Object (...)))
ReflectionException: Property baz does not exist

/usr/src/app/src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php:205
/usr/src/app/src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php:112
/usr/src/app/src/Symfony/Component/PropertyInfo/Tests/Extractor/ReflectionExtractorTest.php:180

15) Symfony\Component\PropertyInfo\Tests\Extractor\ReflectionExtractorTest::testExtractPhp7Type with data set #3 ('buz', array(Symfony\Component\PropertyInfo\Type Object (...)))
ReflectionException: Property buz does not exist

/usr/src/app/src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php:205
/usr/src/app/src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php:112
/usr/src/app/src/Symfony/Component/PropertyInfo/Tests/Extractor/ReflectionExtractorTest.php:180

16) Symfony\Component\PropertyInfo\Tests\Extractor\ReflectionExtractorTest::testExtractPhp7Type with data set #4 ('biz', array(Symfony\Component\PropertyInfo\Type Object (...)))
ReflectionException: Property biz does not exist

/usr/src/app/src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php:205
/usr/src/app/src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php:112
/usr/src/app/src/Symfony/Component/PropertyInfo/Tests/Extractor/ReflectionExtractorTest.php:180

17) Symfony\Component\PropertyInfo\Tests\Extractor\ReflectionExtractorTest::testExtractPhp7Type with data set #5 ('donotexist', null)
ReflectionException: Property donotexist does not exist

/usr/src/app/src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php:205
/usr/src/app/src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php:112
/usr/src/app/src/Symfony/Component/PropertyInfo/Tests/Extractor/ReflectionExtractorTest.php:180

18) Symfony\Component\PropertyInfo\Tests\Extractor\ReflectionExtractorTest::testExtractPhp71Type with data set #0 ('foo', array(Symfony\Component\PropertyInfo\Type Object (...)))
ReflectionException: Property foo does not exist

/usr/src/app/src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php:205
/usr/src/app/src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php:112
/usr/src/app/src/Symfony/Component/PropertyInfo/Tests/Extractor/ReflectionExtractorTest.php:200

19) Symfony\Component\PropertyInfo\Tests\Extractor\ReflectionExtractorTest::testExtractPhp71Type with data set #1 ('buz', array(Symfony\Component\PropertyInfo\Type Object (...)))
ReflectionException: Property buz does not exist

/usr/src/app/src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php:205
/usr/src/app/src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php:112
/usr/src/app/src/Symfony/Component/PropertyInfo/Tests/Extractor/ReflectionExtractorTest.php:200

20) Symfony\Component\PropertyInfo\Tests\Extractor\ReflectionExtractorTest::testExtractPhp71Type with data set #2 ('bar', array(Symfony\Component\PropertyInfo\Type Object (...)))
ReflectionException: Property bar does not exist

/usr/src/app/src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php:205
/usr/src/app/src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php:112
/usr/src/app/src/Symfony/Component/PropertyInfo/Tests/Extractor/ReflectionExtractorTest.php:200

21) Symfony\Component\PropertyInfo\Tests\Extractor\ReflectionExtractorTest::testExtractPhp71Type with data set #3 ('baz', array(Symfony\Component\PropertyInfo\Type Object (...)))
ReflectionException: Property baz does not exist

/usr/src/app/src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php:205
/usr/src/app/src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php:112
/usr/src/app/src/Symfony/Component/PropertyInfo/Tests/Extractor/ReflectionExtractorTest.php:200

22) Symfony\Component\PropertyInfo\Tests\Extractor\ReflectionExtractorTest::testExtractPhp71Type with data set #4 ('donotexist', null)
ReflectionException: Property donotexist does not exist

/usr/src/app/src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php:205
/usr/src/app/src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php:112
/usr/src/app/src/Symfony/Component/PropertyInfo/Tests/Extractor/ReflectionExtractorTest.php:200

ERRORS!
Tests: 188, Assertions: 251, Errors: 22, Skipped: 3.
```

Reply:
You can easily reproduce this tests by running this in `src/Symfony/Component/PropertyInfo`
```bash
docker run --rm -it -v $(pwd):/var/www/html devilbox/php-fpm-7.4 bash
composer install
composer require --dev symfony/test-pack
vendor/bin/simple-phpunit
```

Reply:
Fixed